### PR TITLE
unit test for Table component

### PR DESCRIPTION
**Pull request by Duwon98**
*override test for table component is temperately removed due to generic override function

### CODE REVIEW

**Comment by Duwon98** [comment]
table coverage
table overrides coverage

File: components/table/tests/unit/caption.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import { Caption, Table, Tbody } from '@westpac/table';
+import { GEL } from '@westpac/core';
+import { render } from '@testing-library/react';
+import { overridesTest } from '../../../../helpers/tests/overrides-test.js';
+import { nestingTest } from '../../../../helpers/tests/nesting-test.js';
+import { CaptionProps } from '../../src/Caption';
+import wbc from '@westpac/wbc';
+
+// Todo: generic override function cause error for table component **fix override function for table component
```
Comment (tomjackking) [inline components/table/tests/unit/caption.spec.tsx:9]:
```suggestion
// TODO: Fix overridesTest function to work with table
// There is no overridesTest function for the various table components due to how the overridesTest works,
// creating warnings relating to unexpected <div> as children/parents
```

File: components/table/tests/unit/caption.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { Caption, Table, Tbody } from '@westpac/table';
+import { GEL } from '@westpac/core';
+import { render } from '@testing-library/react';
+import { overridesTest } from '../../../../helpers/tests/overrides-test.js';
```
Comment (tomjackking) [inline components/table/tests/unit/caption.spec.tsx:4]:
```suggestion
```

File: components/table/tests/unit/table.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+import { blenderTable } from '../../src/overrides/table.js';
+import { blenderTd } from '../../src/overrides/td.js';
+import { blenderTr } from '../../src/overrides/tr.js';
+import { overridesTest } from '../../../../helpers/tests/overrides-test.js';
+import { nestingTest } from '../../../../helpers/tests/nesting-test.js';
+import { TableProps } from '../../src/Table';
+import wbc from '@westpac/wbc';
+import { ErrorBoundary } from '../../../../helpers/tests/error-boundary.js';
+
+// Todo: generic override function cause error for table component **fix override function for table component
```
Comment (tomjackking) [inline components/table/tests/unit/table.spec.tsx:13]:
```suggestion
// TODO: Fix overridesTest function to work with table
// There is no overridesTest function for the various table components due to how the overridesTest works,
// creating warnings relating to unexpected <div> as children/parents```

File: components/table/tests/unit/table.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
+import { Table, Caption, Td, Tbody, Tr } from '@westpac/table';
+import { GEL } from '@westpac/core';
+import { render, screen } from '@testing-library/react';
+import { blenderTable } from '../../src/overrides/table.js';
+import { blenderTd } from '../../src/overrides/td.js';
+import { blenderTr } from '../../src/overrides/tr.js';
+import { overridesTest } from '../../../../helpers/tests/overrides-test.js';
```
Comment (tomjackking) [inline components/table/tests/unit/table.spec.tsx:7]:
```suggestion
```

File: components/table/tests/unit/tbody.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import { Tbody, Tr, Td, Table } from '@westpac/table';
+import { GEL } from '@westpac/core';
+import { render } from '@testing-library/react';
+import { overridesTest } from '../../../../helpers/tests/overrides-test.js';
```
Comment (tomjackking) [inline components/table/tests/unit/tbody.spec.tsx:4]:
```suggestion
```

File: components/table/tests/unit/tbody.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import { nestingTest } from '../../../../helpers/tests/nesting-test.js';
+import { TbodyProps } from '../../src/Tbody';
+import wbc from '@westpac/wbc';
+
+// Todo: generic override function cause error for table component **fix override function for table component
```
Comment (tomjackking) [inline components/table/tests/unit/tbody.spec.tsx:9]:
```suggestion
// TODO: Fix overridesTest function to work with table
// There is no overridesTest function for the various table components due to how the overridesTest works,
// creating warnings relating to unexpected <div> as children/parents```

File: components/table/tests/unit/td.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
+import { Td, Tr, Table, Tbody } from '@westpac/table';
+import { GEL } from '@westpac/core';
+import { render } from '@testing-library/react';
+import { overridesTest } from '../../../../helpers/tests/overrides-test.js';
```
Comment (tomjackking) [inline components/table/tests/unit/td.spec.tsx:4]:
```suggestion
```

File: components/table/tests/unit/td.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+import { nestingTest } from '../../../../helpers/tests/nesting-test.js';
+import { TdProps } from '../../src/Td';
+import wbc from '@westpac/wbc';
+
+// Todo: generic override function cause error for table component **fix override function for table component
```
Comment (tomjackking) [inline components/table/tests/unit/td.spec.tsx:9]:
```suggestion
// TODO: Fix overridesTest function to work with table
// There is no overridesTest function for the various table components due to how the overridesTest works,
// creating warnings relating to unexpected <div> as children/parents```

File: components/table/tests/unit/tfoot.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
+import { Tfoot, Tr, Table, Td } from '@westpac/table';
+import { GEL } from '@westpac/core';
+import { render } from '@testing-library/react';
+import { overridesTest } from '../../../../helpers/tests/overrides-test.js';
```
Comment (tomjackking) [inline components/table/tests/unit/tfoot.spec.tsx:4]:
```suggestion
```

File: components/table/tests/unit/tfoot.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+import { nestingTest } from '../../../../helpers/tests/nesting-test.js';
+import { TfootProps } from '../../src/Tfoot';
+import wbc from '@westpac/wbc';
+
+// Todo: generic override function cause error for table component **fix override function for table component
```
Comment (tomjackking) [inline components/table/tests/unit/tfoot.spec.tsx:9]:
```suggestion
// TODO: Fix overridesTest function to work with table
// There is no overridesTest function for the various table components due to how the overridesTest works,
// creating warnings relating to unexpected <div> as children/parents
```

Comment (tomjackking) [inline components/table/tests/unit/tfoot.spec.tsx:9]:
```suggestion
// TODO: Fix overridesTest function to work with table
// There is no overridesTest function for the various table components due to how the overridesTest works,
// creating warnings relating to unexpected <div> as children/parents
```

File: components/table/tests/unit/th.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,50 @@
+import { Th, Table, Tr, Thead } from '@westpac/table';
+import { GEL } from '@westpac/core';
+import { render } from '@testing-library/react';
+import { overridesTest } from '../../../../helpers/tests/overrides-test.js';
```
Comment (tomjackking) [inline components/table/tests/unit/th.spec.tsx:4]:
```suggestion
```

File: components/table/tests/unit/th.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+import { nestingTest } from '../../../../helpers/tests/nesting-test.js';
+import { ThProps } from '../../src/Th';
+import wbc from '@westpac/wbc';
+
+// Todo: generic override function cause error for table component **fix override function for table component
```
Comment (tomjackking) [inline components/table/tests/unit/th.spec.tsx:9]:
```suggestion
// TODO: Fix overridesTest function to work with table
// There is no overridesTest function for the various table components due to how the overridesTest works,
// creating warnings relating to unexpected <div> as children/parents
```

File: components/table/tests/unit/thead.spec.tsx [hunk 13]
```diff
@@ -0,0 +1,45 @@
+import { Thead, Table, Tr } from '@westpac/table';
+import { GEL } from '@westpac/core';
+import { render } from '@testing-library/react';
+import { overridesTest } from '../../../../helpers/tests/overrides-test.js';
```
Comment (tomjackking) [inline components/table/tests/unit/thead.spec.tsx:4]:
```suggestion
import { overridesTest } from '../../../../helpers/tests/overrides-test.js';
```

Comment (tomjackking) [inline components/table/tests/unit/thead.spec.tsx:4]:
```suggestion
```

File: components/table/tests/unit/thead.spec.tsx [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+import { nestingTest } from '../../../../helpers/tests/nesting-test.js';
+import { TheadProps } from '../../src/Thead';
+import wbc from '@westpac/wbc';
+
+// Todo: generic override function cause error for table component **fix override function for table component
```
Comment (tomjackking) [inline components/table/tests/unit/thead.spec.tsx:9]:
```suggestion
// TODO: Fix overridesTest function to work with table
// There is no overridesTest function for the various table components due to how the overridesTest works,
// creating warnings relating to unexpected <div> as children/parents```

Comment (tomjackking) [inline components/table/tests/unit/thead.spec.tsx:9]:
```suggestion
// TODO: Fix overridesTest function to work with table
// There is no overridesTest function for the various table components due to how the overridesTest works,
// creating warnings relating to unexpected <div> as children/parents
```

File: components/table/tests/unit/tr.spec.tsx [hunk 15]
```diff
@@ -0,0 +1,74 @@
+import { Tr, Th, Td, Thead, Table } from '@westpac/table';
+import { GEL } from '@westpac/core';
+import { render } from '@testing-library/react';
+import { overridesTest } from '../../../../helpers/tests/overrides-test.js';
```
Comment (tomjackking) [inline components/table/tests/unit/tr.spec.tsx:4]:
```suggestion
```

File: components/table/tests/unit/tr.spec.tsx [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+import { nestingTest } from '../../../../helpers/tests/nesting-test.js';
+import { TrProps } from '../../src/Tr';
+import wbc from '@westpac/wbc';
+
+// Todo: generic override function cause error for table component **fix override function for table component
```
Comment (tomjackking) [inline components/table/tests/unit/tr.spec.tsx:9]:
```suggestion
// TODO: Fix overridesTest function to work with table
// There is no overridesTest function for the various table components due to how the overridesTest works,
// creating warnings relating to unexpected <div> as children/parents```

**Review by soobin93 — APPROVED** [review]

**Review by tomjackking — APPROVED** [review]

**Pull request merged**